### PR TITLE
Add option to control timeout for Basic Auth

### LINKED ISSUE #210: Basic auth expires after 10 minutes
Hi everyone,

I'm using this module to authenticate the users with their AD credentials via kerberos, basic auth and a form login (which authenticates the user via PHP + basic auth). The `GssapiUseSessions` option is set to `On`, so the authentication process needs to be done only once. However, if I use basic auth for the first login and the session cookie for the following requests, the user can't be authenticated anymore after exactly 10 minutes. 

According to the source code, the user should be authenticated *indefinite* and therefore logged in as long as the session is alive. The default `krb5.conf` should obtain tickets for one day. However, the log states 

> NO AUTH DATA Client did not send any authentication headers 

and the request is terminated with an error 401.

So I guess the ticket is only valid for 10 minutes instead of a day. Is there any chance to increase this time limit? I couldn't find any hint in the source code, so I would appreciate some help.

Thanks ;)

### PR DESCRIPTION
**Pull request by simo5**
Fixes #210

### CODE REVIEW

File: src/mod_auth_gssapi.c [hunk 1]
```diff
@@ -1808,6 +1809,21 @@ static const char *mag_acceptor_name(cmd_parms *parms, void *mconfig,
...
 }
 
+static const char *mag_basic_timeout(cmd_parms *parms, void *mconfig,
+                                     const char *w)
+{
+    struct mag_config *cfg = (struct mag_config *)mconfig;
+    long value;
+
+    value = strtol(w, NULL, 10);
+    if (value <= 0 || value >= UINT32_MAX) {
```
Comment (frozencemetery) [inline src/mod_auth_gssapi.c:1819]: `value` is signed here - you need to check against INT32_MAX here as that's smaller.

File: src/sessions.c [hunk 2]
```diff
@@ -161,6 +161,7 @@ void mag_check_session(struct mag_req_cfg *cfg, struct mag_conn **conn)
         mc->established = false;
         goto done;
     }
+fprintf(stderr, "EXPIRE: %ld, TIME: %ld\n", expiration, time(NULL));
```
Comment (frozencemetery) [inline src/sessions.c:164]: Indentation here is wrong.

File: tests/t_basic_timeout.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+# Copyright (C) 2015 - mod_auth_gssapi contributors, see COPYING for license.
```
Comment (frozencemetery) [inline tests/t_basic_timeout.py:2]: 2020

File: src/mod_auth_gssapi.c (hunk 1 above)
Comment (simo5) [reply src/mod_auth_gssapi.c:1819]: value is a long which should be bigger than uint32 anyway ?
I can also use long long and strtoll() if there are platforms where long is not at least 64bit ..

File: src/sessions.c (hunk 2 above)
Comment (simo5) [reply src/sessions.c:164]: you caught a debug message that I forgot to remove :-D

File: tests/t_basic_timeout.py (hunk 3 above)
Comment (simo5) [reply tests/t_basic_timeout.py:2]: time is overrated :-D
fixing ...

File: src/mod_auth_gssapi.c (hunk 1 above)
Comment (simo5) [reply src/mod_auth_gssapi.c:1820]: @frozencemetery what do you prefer ?

Comment (frozencemetery) [reply src/mod_auth_gssapi.c:1820]: I mean, why not just use LONG_MAX (the man page suggests this)?

Technically, C specifies (in C99, 5.2.4.2.1, pages 21-22) that a long has to be at least 32 bits, and an integer 16 bits.  It's permitted for sizeof(long) == sizeof(int) - in fact, this is the case on 32-bit x86:

```
rharwood@seton:/tmp$ cat test.c
#include <stdio.h>

int main() {
    printf("long: %ud, int: %ud\n", sizeof(long), sizeof(int));
    return 0;
}
rharwood@seton:/tmp$ gcc -m32 test.c
rharwood@seton:/tmp$ gcc -m32 -o test test.c
rharwood@seton:/tmp$ ./test 
long: 4d, int: 4d
rharwood@seton:/tmp$ 
```

Comment (simo5) [reply src/mod_auth_gssapi.c:1820]: I guess I should actually simply use strtoul, and then the problem resolves itself.
I'll change it that way.

**Pull request merged**